src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,210 @@
+        if (!queryField.isDisabled()) {
+            queryField.setDisable(true);
+        }
+        finalAssignedLabels = updateFinalAssignedLabels(label);
+        populatePanes(new ArrayList<>(), Optional.empty());
+    }
+
+    // Initilisation of UI
 
-        Platform.runLater(textField::requestFocus);
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        registerQueryHandler();
     }
 
-    private void setupEvents(Stage stage) {
-        setupKeyEvents();
+    private void initUI(Stage stage, TurboIssue issue) {
+        initialiseDialog(stage, issue);
+        setDialogPaneContent(issue);
+        title.setTooltip(createTitleTooltip(issue));
+        createButtons();
+        
+        populatePanes(new ArrayList<>(), Optional.empty());
+    }
 
-        showingProperty().addListener(e -> {
-            positionDialog(stage);
-        });
+    private void initialiseDialog(Stage stage, TurboIssue issue) {
+        initOwner(stage);
+        initModality(Modality.APPLICATION_MODAL); 
+        setTitle("Edit Labels for " + (issue.isPullRequest() ? "PR #" : "Issue #") +
+                issue.getId() + " in " + issue.getRepoId());
     }
 
-    private void setupKeyEvents() {
-        textField.textProperty().addListener((observable, oldValue, newValue) -> {
-            uiLogic.processTextFieldChange(newValue.toLowerCase());
-        });
-        textField.setOnKeyPressed(e -> {
-            if (!e.isAltDown() && !e.isMetaDown() && !e.isControlDown()) {
-                if (e.getCode() == KeyCode.DOWN) {
-                    e.consume();
-                    uiLogic.moveHighlightOnLabel(true);
-                } else if (e.getCode() == KeyCode.UP) {
-                    e.consume();
-                    uiLogic.moveHighlightOnLabel(false);
-                } else if (e.getCode() == KeyCode.SPACE) {
-                    e.consume();
-                    uiLogic.toggleSelectedLabel(textField.getText());
-                }
-            }
-        });
+    private void setDialogPaneContent(TurboIssue issue) {
+        createMainLayout();
+        setTitleLabel(issue);
+        getDialogPane().setContent(mainLayout);
     }
 
-    @SuppressWarnings("unused")
-    private void ______POPULATION______() {}
+    // Population of UI elements
 
-    protected void populatePanes(List<PickerLabel> existingLabels, List<PickerLabel> newTopLabels,
-            List<PickerLabel> bottomLabels, Map<String, Boolean> groups) {
-        populateTopPane(existingLabels, newTopLabels);
-        populateBottomBox(bottomLabels, groups);
+    /**
+     * Updates ui elements based on current state
+     * @param state
+     */
+    public void populatePanes(List<String> matchedLabels, Optional<String> suggestion) {
+        // Population of UI elements
+        populateAssignedLabels(finalAssignedLabels, suggestion);
+        populateFeedbackLabels(finalAssignedLabels, matchedLabels, suggestion);
```

Comment:
The comment is a bit obvious. :stuck_out_tongue: Also the javadoc comment is outdated. Does `finalAssignedLabels` need to be passed as a parameter?